### PR TITLE
add cached readArg1Str()

### PR DESCRIPTION
**Pull request by Raynos**
This is the third & final method to allow reading of
arg1 efficiently.

Going to add benchmarks whilst waiting for reviews.

r: @jcorbin @rf

### CODE REVIEW

**Comment by rf** [comment]
lgtm

**Comment by Raynos** [comment]
Ran the benchmarks.

readArg1Str() is about 4x faster.

All 4 methods together is about 3x faster.

File: v2/call.js [hunk 1]
```diff
@@ -298,6 +298,48 @@ function readUInt8String(frame, offset) {
...
+
+    var offset = frame.cache.csumStartOffset;
+
+    if (frame.size < offset + 1) {
+        return null;
+    }
+    var csumType = frame.buffer.readUInt8(offset, false);
+    offset += 1;
+
+    if (csumType !== 0x00) {
```
Comment (jcorbin) [inline v2/call.js]: please use the constant

File: v2/call.js [hunk 2]
```diff
@@ -298,6 +298,48 @@ function readUInt8String(frame, offset) {
...
+        offset += 4;
```
Comment (jcorbin) [inline v2/call.js]: please use checksum utility function or simple lookup table by type for this, don't hardcode

**Pull request merged**